### ISSUE TITLE
Consider resetting xDomainMessageSender after relaying a message 

### ISSUE THREAD

**Issue by cwhinfrey** [ethereum-optimism/contracts#309]
**Is your feature request related to a problem? Please describe.**
`xDomainMessageSender` will continue to return the sender of the last message until the next message is relayed.

This shouldn't be an issue for the most part since `relayMessage` has a reentrancy guard and every contract consuming `xDomainMessageSender` should also be checking that `msg.sender` is the Optimism messenger. That being said, it could lead to some weird behavior and is easy to fix.

**Describe the solution you'd like**
Consider either zeroing out xDomainMessageSender after the calls [here](https://github.com/ethereum-optimism/contracts/blob/master/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol#L117) and [here](https://github.com/ethereum-optimism/contracts/blob/master/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L2CrossDomainMessenger.sol#L79) or setting is to an address like `0x000000000000000000000000000000000000dead`.  Either one should result in some [gas savings](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2200.md) since the transaction will always reset the storage slot to the same value that it was at the beginning of the transaction. Each option will have slightly different [gas implications](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/security/ReentrancyGuard.sol#L28-L32) (I'd recommend setting to the dead address).

Basically, it should improve security and save a bit of gas.

**Describe alternatives you've considered**
Leave as is
